adaptdl/adaptdl/torch/data.py

```diff
@@ -276,12 +276,40 @@ def _sync_local_bsz(self):
         else:
             # Autoscale batch size, compute on rank 0 and broadcast.
             speedup_fn = get_speedup_fn()
-            _, (local_bsz, accumulation_steps) = speedup_fn(
-                adaptdl.env.num_nodes(),
-                adaptdl.env.num_replicas(),
-                return_config=True)
-            (self._current_local_bsz, self._accumulation_steps) = \
-                adaptdl.collective.broadcast((local_bsz, accumulation_steps))
+
+            # check if batch size  is up
+            is_init = self._current_local_bsz is None
+            # if init, use the batch size suggested
+            if is_init:
+                _, (local_bsz, accumulation_steps) = speedup_fn(
+                    adaptdl.env.num_nodes(),
+                    adaptdl.env.num_replicas(),
+                    return_config=True)
+                (self._current_local_bsz, self._accumulation_steps) = \
+                    adaptdl.collective.broadcast((local_bsz,
+                                                  accumulation_steps))
+            # if not first time, we check against the relative speedup
+            else:
+                suggest_speedup, (local_bsz, accumulation_steps) = speedup_fn(
+                                   adaptdl.env.num_nodes(),
+                                   adaptdl.env.num_replicas(),
+                                   return_config=True)
+                # get current speedup
+                current_speedup = speedup_fn(adaptdl.env.num_nodes(),
+                                             adaptdl.env.num_replicas(),
+                                             local_bsz=self.current_local_bsz)
+            # use only if speedup is significant
+                speedup_to_cur = suggest_speedup / current_speedup
+                if speedup_to_cur > self.speedup_threshold:
+                    (self._current_local_bsz, self._accumulation_steps) = \
+                        adaptdl.collective.broadcast((local_bsz,
+                                                      accumulation_steps))
+                else:
+                    (self._current_local_bsz, self._accumulation_steps) = \
+                        adaptdl.collective.broadcast((
+                            self._current_local_bsz,
+                            self._accumulation_steps))
```

Comment:
It might be better to run
```
self._current_local_bsz, self._accumulation_steps) = \
    adaptdl.collective.broadcast((
        self._current_local_bsz,
        self._accumulation_steps))
```
just after all the if-else branches, and simply assign `self._current_local_bsz` and `self._accumulation_steps` in the if-else branches. That way it's a little less verbose and makes it clear that the broadcast is always called exactly once on each replica.